Scraper Scripts/scraper.js

```diff
@@ -1,129 +1,741 @@
+                await page.click("button#search");
+                await sleep.setTimeout(250);
+                for (let j = 0; j < 1000; j++) {
+                    await page.keyboard.press("Backspace");
+                }
+                await page.type("input#cli_shellHeaderSearchInput", gameTitles[i], {delay: 300});
+
+                await page.keyboard.press("Enter");
+                await page.waitForNavigation(); // Loads the search results and waits for it to navigate there
+                let gameDataList;
+
+                await page.waitForSelector("div.ProductCard-module__cardWrapper___6Ls86 > a > div + div", {timeout: 3000});
+                await sleep.setTimeout(1500);
+                
+                // Returns an array of objects containing all the data we can scrape from Xbox. They don't have a display for deal end
+                gameDataList = await page.evaluate((gameTitle) => {
+                    let itemsList = document.querySelectorAll("div.ProductCard-module__infoBox___M5x18");
+                    let dataList = [];
+                    let j = 0;
+                    itemsList.forEach(element => {
+                        if (element.innerText.includes(gameTitle)) {
+                            let data = {
+                                "Title": element.querySelector("span").textContent,
+                                "OriginalPrice": element.querySelector("div > div > span")?.textContent.substring(5) ?? element.querySelector("span + span")?.textContent,
+                                "DiscountPrice": element.querySelector("div > div > span + span")?.textContent.substring(5) ?? element.querySelector("div > div > span")?.textContent.substring(5) ?? element.querySelector("span + span")?.textContent,
+                                "DiscountPerc": element.querySelector("div > div + div")?.textContent ?? "0%"
+                            };
+                            dataList[j++] = data;
+                        }
+                    });
+                    return dataList;
+                }, gameTitles[i]);
+
+                await sleep.setTimeout(1000);
+                let gameData = [];
+                gameData.length = gameDataList.length;
+                for (let i = 0; i < gameDataList.length; i++) {
+                    gameData[i] = { // Processing all the data into a new array to reduce complexity below
+                        "Title": gameDataList[i]["Title"],
+                        "OriginalPrice": GetPriceFromString(gameDataList[i]["OriginalPrice"]),
+                        "DiscountPrice": GetPriceFromString(gameDataList[i]["DiscountPrice"]),
+                        "DiscountPerc": gameDataList[i]["DiscountPerc"]
+                    }
+                }
+
+                if (gameData.length > 1) { // If we got more than 1 version of a game, map the real index of the whole
+                    // data to an incremented index of our gathered data
```

Comment:
nit: you'll have more space for a comment if you put it above
```suggestion
                // If we got more than 1 version of a game, map the real index of the whole data to an incremented index of our gathered data
                if (gameData.length > 1) { 
```